assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { trackEvent } from '@web-stories-wp/tracking';
+import { useEffect, useRef, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../../app';
+import { ChecklistCard, DefaultFooterText } from '../../checklistCard';
+import { PRIORITY_COPY } from '../constants';
+import { useRegisterCheck } from '../countContext';
+
+export async function getStoryAmpValidationErrors({ link, status }) {
+  // TODO: Enable for drafts as well?
+  if (!link || !['publish', 'future'].includes(status)) {
+    return false;
+  }
+
+  try {
+    const response = await fetch(link);
+    const storyMarkup = await response.text();
+    const { status: markupStatus, errors } =
+      window.amp.validator.validateString(storyMarkup);
+
+    if ('FAIL' !== markupStatus) {
+      return false;
+    }
+
+    const filteredErrors = errors
+      .filter(({ severity }) => severity === 'ERROR')
+      .filter(({ code, params }) => {
+        // Filter out errors that are covered in other checks
+        // Already covered by metadata checks.
+        if ('MISSING_URL' === code && params[0].startsWith('poster')) {
+          return false;
+        }
```

Comment:
We probably need to add more here, for example to filter out the publisher logo warnings.

Reply:
I'll look closer at the AMP documentation today and try to filter out some more. Asked @choumx to take a look as he's familiar with AMP.

Reply:
I think filtering on a validation error-by-error basis will be difficult and probably brittle.

Could we make triggering of the entire check conditional instead? If we only trigger the AMP validation PPC when there are no high priority PPC items, then we know that any validation errors found are not duplicative.

Reply:
🤔  

so, that would mean the following set of actions/responses
- story reaches checkpoint so priority are visible (publish attempt or more than 4 pages)
- I fix all my priority issues 
- amp validation error pops up *is this a priority issue or a new section*?
- If a high priority item returns (i don't know know why you'd do this but let's say you remove your title), does the amp validation error message disappear?

Reply:
There are only 4 priority checks related to issues that would cause AMP invalidation.

* Missing title
* Missing story poster
* Missing publisher logo
* Missing video posters

Filtering by that seems pretty straightforward.

Reply:
so that'll work if we can move this check to be anywhere besides the priority panel. It gets goofy with the logic if we are trying to render something in the priority panel that isn't counted as a priority issue and should only show up when the priority panel has been resolved and otherwise wouldn't be open.

Thoughts on it existing in the accessibility panel? 
![Screen Shot 2021-07-27 at 2 27 29 PM](https://user-images.githubusercontent.com/10720454/127229840-31da33d0-f0c8-46ef-ac3f-d39a5359eead.png)

Reply:
That sounds like an odd category for this check. Perhaps a new one is needed? e.g. „Distribution“ or „Validation“

Reply:
So testing this a bit more and I'm a bit confused. 

If you take a template and make a story out of it and clear up the priority issues, you'll see the new amp error check show up  and it has the following codes causing it to show up: `MISSING_REQUIRED_EXTENSION`, `DISALLOWED_SCRIPT_TAG`, `WRONG_PARENT_TAG`, `MANDATORY_TAG_MISSING` but they are all resolved the moment you save the story or publish it.


This leads me to think that we should only check published stories, not drafts as it's not a relevant issue to show to people until they have published. Alternately, could track for first save and start showing it then, but it feels like a check that should happen once the story is done.

https://user-images.githubusercontent.com/10720454/127231857-0797855c-6b9c-4236-befa-9e6c52c8123b.mp4



Thoughts @choumx @swissspidy ?

Reply:
> That sounds like an odd category for this check. Perhaps a new one is needed? e.g. „Distribution“ or „Validation“

Agree, @swissspidy - let's do distribution

Reply:
So the video above is still applicable, just pretend it shows up in this new panel. I think this is something that only needs to show up on publish because most of these errors are resolving on save, so they'd only be relevant if they stick around after publishing. 

Am i on the right track? 

![Screen Shot 2021-07-27 at 3 01 18 PM](https://user-images.githubusercontent.com/10720454/127233280-15982bb3-7984-474b-a3f0-8742a161a186.png)

Reply:
also, @choumx or @o-fernandez please someone make that copy better than `Make your Web Story distributable`

Reply:
> There are only 4 priority checks related to issues that would cause AMP invalidation.... Filtering by that seems pretty straightforward. 

Hmm, we can still try filtering errors if the performance of validation is acceptable. A nice thing about limiting triggering (e.g. 0 priority items and on-publish) is that it helps constrain performance concerns.

Regardless, we should highlight this check so that users are aware of issues before publishing. That means making sure it's treated the same as high priority items in the blocking popup when clicking publish, etc.

(I can even imagine an alternative UX where this check **only** lives in that blocking popup instead of as a new PPC item -- would that be simpler?) 

> If a high priority item returns (i don't know know why you'd do this but let's say you remove your title), does the amp validation error message disappear?

I agree this seems a bit weird but maybe it's OK since this AMP validation check should just be a rare catch-all for configuration/environment issues like conflicting plugins, etc.

This also should inform the copy we use -- IMO we should:
- Suggest possible issues with the creator's WP setup (in the check or in the FAQ/help article)
- Link to the FAQ or forums instead of the AMP Test Tool (which was built for developers)
- Consider avoiding developer terms like "validation" ("compatibility" instead?)

> Fix compatibility errors to make your story eligible to appear on Google.
> ### Compatibility
> Your story contains errors that can affect eligibility to appear on Google. This can be caused by incompatible WordPress plugins, web application firewalls, or other hosting configuration issues. Please see [our FAQ](https://wordpress.org/plugins/web-stories/#what%20do%20i%20do%20about%20an%20amp%20validation%20issue%20in%20search%20console%3F) for more information and debugging tips.

Reply:
> This leads me to think that we should only check published stories, not drafts as it's not a relevant issue to show to people until they have published.

That sounds fine to me too.

Reply:
> Hmm, we can still try filtering errors if the performance of validation is acceptable. A nice thing about limiting triggering (e.g. 0 priority items and on-publish) is that it helps constrain performance concerns. 

the way I've got it now is the validation check will only run once there are no priority issues left, additionally it's only ever updating when `isSaved` is triggered so i'd say performance isn't something we need to be really worried about on this one here. 

> we should highlight this check so that users are aware of issues before publishing. That means making sure it's treated the same as high priority items in the blocking popup when clicking publish, etc. 

This is going to be more involved than just adding a new check. 

It _feels like_ this is a priority check. I'm hesitant to stick with a new panel. If it's really only 4 priority checks that we need to check and see if they exist before rendering the AMP check, then I think just checking internally how we had it earlier would be the way to go. Since the check's memoized and only running on `isSaved` it's not contingent on the entire story updating all the time. It wouldn't be a big deal, perf wouldn't be impacted. That would make sure users have the same shot at reviewing it as other priority issues. 

Thoughts? Given our approaching release we should either move this to the next release if we want its own panel with new dialog logic and triggers or we should put it in the priority panel and limit the check internally which i can wrap up tomorrow.

Reply:
If you want to jump on a call in the AM we can work it out that way too.

Reply:
ok dokey! 

@choumx @swissspidy  I updated. Can you take a peak?  I updated the PR description to be central to this thread and the decisions made. It ended up being just barely different from where we started yesterday AM 😅. 

### TL;DR
- AMP Validation Error check is part of priority group
- Filtering out for known priority issues (publisher logo, story poster, video poster, title (this one happens naturally it seems? I couldn't get it to fire, if you can awesome.)) 
- Check will only ever update 1) initial render 2) whenever `isSaving` becomes false again, this means that any issues that could have been present but would be cleaned up on a save or publish are now gone so it silences some unnecessary noise. 
 _A note on the above_: This is something you can see happen if you create a story from a template and unblock the validation check for drafts/save, but when you publish the template story then they all go away. 
- since it's included in priority check, if a story is scheduled or published and there's some errors hanging on then the little exclamation icon will stay present with the tooltip to remind user to check it out (or if they haven't seen the checklist at all, that dialog will show). 

Anyways, this is now behind a feature flag (defaulted to true) and we've got some event tracking set up. I'm curious how often this even fires? Feels like a good first implementation. 

Let me know what you think and [if the wording](https://github.com/google/web-stories-wp/pull/8443#discussion_r678676306) is appropriate or if there's other specific filters we should add to the check that I am unaware of. 

Thanks!

Reply:
> title (this one happens naturally it seems? I couldn't get it to fire, if you can awesome.

I think AMP validation checks for existence of the `amp-story[title]` attribute while our PPC check looks for empty string:

https://github.com/google/web-stories-wp/blob/d093d9f0980d4a3b2a8c5ed2a669308a95993eb9/assets/src/edit-story/output/story.js#L74-L80